cmake/emu/nsim.cmake

```diff
@@ -1,5 +1,27 @@
 # SPDX-License-Identifier: Apache-2.0
+if(${CONFIG_SOC_NSIM_HS_SMP})
+# mdb is required to run nsim multicore targets
+find_program(
+  MDB
+  mdb
+  )
+
+if(${CONFIG_SOC_NSIM_HS_SMP})
+ set(MDB_ARGS mdb_hs_smp.args)
+endif()
 
+add_custom_target(run
+  COMMAND
+  ${MDB} -pset=1 -psetname=core0 -prop=ident=0x00000050 -cmpd=soc
+  @${BOARD_DIR}/support/${MDB_ARGS} ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME} &&
+  ${MDB} -pset=2 -psetname=core1 -prop=download=2 -prop=ident=0x00000150 -cmpd=soc
+  @${BOARD_DIR}/support/${MDB_ARGS} ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME} &&
```

Comment:
I know this is old code being moved, but this: `${APPLICATION_BINARY_DIR}/zephyr` should have been `${ZEPHYR_BINARY_DIR}`.